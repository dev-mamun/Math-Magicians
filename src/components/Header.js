/** ****************************************
 * Project: math-magicians
 * File: Header.js
 * Created: 4/7/23
 * Author: Abdullah Al Mamun <mamun1214@gmail.com>
 ****************************************** */
import { Link, NavLink } from 'react-router-dom';

export default function Header() {
  return (
    <>
      <nav className="navbar navbar-expand-lg bg-light mb-3 px-2" aria-label="Thirteenth navbar">
        <button
          className="navbar-toggler"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbarSupportedContent"
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse " id="navbarSupportedContent">
          <a
            className="navbar-brand col-lg-3"
            href="/"
          >
            Math Magicians
          </a>
          <ul className="navbar-nav mb-2 mb-lg-0  ms-auto">
            <li className="nav-item">
              <NavLink className="nav-link active" to="/">Home</NavLink>
            </li>
            <li className="nav-item">
              <Link className="nav-link" to="/calculator">Calculator</Link>
            </li>
            <li className="nav-item">
              <Link className="nav-link" to="/quotes">Quote</Link>
            </li>
          </ul>
        </div>
      </nav>
    </>
  );
}
