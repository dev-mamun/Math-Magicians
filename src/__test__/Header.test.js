/** ****************************************
 * Project: math-magicians
 * File: Header.test.js
 * Created: 4/8/23
 * Author: Abdullah Al Mamun <mamun1214@gmail.com>
 ****************************************** */
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import Header from '../components/Header';

it('Header Render Correctly', () => {
  const tree = render(<BrowserRouter><Header /></BrowserRouter>);
  expect(tree).toMatchSnapshot();
});

it('Header UI', () => {
  render(
    <BrowserRouter>
      <Header />
    </BrowserRouter>,
  );
  const Element = screen.getByText('Math Magicians');
  expect(Element).toBeInTheDocument();
});
