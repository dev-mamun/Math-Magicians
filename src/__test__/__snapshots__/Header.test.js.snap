// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Render Correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        aria-label="Thirteenth navbar"
        class="navbar navbar-expand-lg bg-light mb-3 px-2"
      >
        <button
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-bs-target="#navbarSupportedContent"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse "
          id="navbarSupportedContent"
        >
          <a
            class="navbar-brand col-lg-3"
            href="/"
          >
            Math Magicians
          </a>
          <ul
            class="navbar-nav mb-2 mb-lg-0  ms-auto"
          >
            <li
              class="nav-item"
            >
              <a
                aria-current="page"
                class="nav-link active active"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/calculator"
              >
                Calculator
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/quotes"
              >
                Quote
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      aria-label="Thirteenth navbar"
      class="navbar navbar-expand-lg bg-light mb-3 px-2"
    >
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse "
        id="navbarSupportedContent"
      >
        <a
          class="navbar-brand col-lg-3"
          href="/"
        >
          Math Magicians
        </a>
        <ul
          class="navbar-nav mb-2 mb-lg-0  ms-auto"
        >
          <li
            class="nav-item"
          >
            <a
              aria-current="page"
              class="nav-link active active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/calculator"
            >
              Calculator
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/quotes"
            >
              Quote
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
